agent/agent_worker.go

```diff
@@ -149,10 +154,6 @@ func (a *AgentWorker) Start() error {
 					a.logger.Error("Failed to heartbeat %s. Will try again in %s. (Last successful was %v ago)",
 						err, heartbeatInterval, time.Now().Sub(lastHeartbeat))
 				}
-
-			case <-a.stop:
```

Comment:
The other way to do this would be with a context with cancel:

```go
ctx, cancel := context.WithCancel(context.Background()
defer cancel()
...
case <-ctx.Done():
```